src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal.js

```diff
@@ -336,7 +350,11 @@ export const CloudEnvironmentModal = ({
     const app = currentApp(toolLabel)
     const doesCloudEnvForToolExist = currentRuntimeTool === toolLabel || app
     const isCloudEnvForToolDisabled = isCloudEnvModalDisabled(toolLabel)
+    // for Public Preview disable the option to create a new Cromwell app in Azure
+    const isSettingsOptionDisabled = toolLabel === toolLabels.CromwellOnAzure
     return h(Fragment, [
+      // We cannot attach the periodic cookie setter until we have a running Cromwell app for Azure because the relay is not guaranteed to be ready until then
+      toolLabel === toolLabels.CromwellOnAzure && app?.status === 'RUNNING' ? h(PeriodicAzureCookieSetter, { proxyUrl: app.proxyUrls['cbas-ui'], forCromwell: true }) : null,
```

Comment:
Isn't the landing zone relay being used? If so, it should be ready before any app could be created.

Reply:
The way it works is that a landing zone is created for a billing project. A billing project can't be used to create a workspace until the the landing zone has completed its creation (this is asynchronous, but the billing project is "spinning" and cannot be selected until the the landing zone completion is done). Therefore by the time a workspace is created, the relay should be up and functioning… if not, it's a but in the landing zone code because they changed the state too early into "DONE". 

I don't think it makes sense to put in defensive checks that would be hiding other bugs… it just adds confusion. 

But this is all assuming that Cromwell/Leo uses the landing zone relay. @Qi77Qi should be able to answer that.

Reply:
This is based on what is currently [implemented for Notebooks on Azure](https://github.com/DataBiosphere/terra-ui/blob/53a25c92debb5b77605f984b483b5f9b34cf7194/src/pages/workspaces/workspace/analysis/AppLauncher.js#L228-L229).

Reply:
@n8landolt @Qi77Qi can you comment on this?

Reply:
@Qi77Qi can you please provide some feedback?